src/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import { ProgressStatus } from './types/ProgressEnum';
+import { Todo } from './types/Todo';
+
+export function getTodosToRender(
+  todos: Todo[],
+  searchQuery: string,
+  progress: ProgressStatus,
+): Todo[] {
+  const preparedQuery = searchQuery.toLowerCase();
+
+  return todos.filter(todo => {
+    switch (progress) {
+      case ProgressStatus.Completed:
+        return todo.completed;
+
+      case ProgressStatus.Active:
+        return !todo.completed;
+
+      default:
+        return true;
+    }
+  })
+    .filter(todo => todo.title.toLowerCase().includes(preparedQuery));
```

Comment:
Calling two filters in a row is not an optimized solution, it is better to combine it into one filter:
```suggestion
  function includesIgnoreCase(str, query) {
    return str.toLowerCase().includes(query);
  }

  return todos.filter(todo => {
    switch (progress) {
      case ProgressStatus.Completed:
        return todo.completed && includesIgnoreCase(todo.title, preparedQuery);
  
      case ProgressStatus.Active:
        return !todo.completed && includesIgnoreCase(todo.title, preparedQuery);
  
      default:
        return includesIgnoreCase(todo.title, preparedQuery);
    }
  });
```